sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/request/GlobalThroughputRequestController.java

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.throughputControl.controller.request;
+
+import com.azure.cosmos.implementation.GlobalEndpointManager;
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.throughputControl.ThroughputRequestThrottler;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URI;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public class GlobalThroughputRequestController implements IThroughputRequestController {
+    private final GlobalEndpointManager globalEndpointManager;
+    private final AtomicReference<Double> scheduledThroughput;
+    private final ConcurrentHashMap<URI, ThroughputRequestThrottler> requestThrottlerMapByRegion;
+
+    public GlobalThroughputRequestController(GlobalEndpointManager globalEndpointManager, double initialScheduledThroughput) {
+        checkNotNull(globalEndpointManager, "Global endpoint manager can not be null");
+
+        this.globalEndpointManager = globalEndpointManager;
+        this.scheduledThroughput = new AtomicReference<>(initialScheduledThroughput);
+        this.requestThrottlerMapByRegion = new ConcurrentHashMap<>();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T> Mono<T> init() {
+        return Flux.fromIterable(this.globalEndpointManager.getReadEndpoints())
+            .mergeWith(Flux.fromIterable(this.globalEndpointManager.getWriteEndpoints()))
```

Comment:
why do we need to merge both write endpoint with read endpoints?

isn't read-endpoints superset of write endpoints?

Reply:
Just feel safer to include both and do not think it will hurt this way, will remove the merge